ASSIGMENT/src/pages/dl.astro

```diff
@@ -0,0 +1,271 @@
+
+            const updateEntry = (item, documentNumber, holdingPersonName, DOB, gender, DOE) => {
+                const maleSymbol = String.fromCharCode(0x2642);
+                const femaleSymbol = String.fromCharCode(0x2640);
+                const genderSymbol = gender === "male" ? maleSymbol : femaleSymbol;
+
+                item.querySelector("div:nth-child(3)").textContent = documentNumber;
+                item.querySelector("div:nth-child(4)").textContent = holdingPersonName;
+                item.querySelector("div:nth-child(5)").textContent = genderSymbol;
+                item.querySelector("div:nth-child(6)").textContent = formatDate(DOB);
+                item.setAttribute("data-DOE",DOE);
+
+                currentEditItem = null;
+            };
+
+            const resetForm = () => {
+                dlForm.reset();
+                currentEditItem = null;
+            };
+
+            const deleteItem = (event) => {
+                const item = event.target.closest(".item");
+                item.remove();
+            };
+
+            const editItem = (event) => {
+                const item = event.target.closest(".item");
+                const documentNumber = item.querySelector("div:nth-child(3)").textContent.trim();
+                const holdingPersonName = item.querySelector("div:nth-child(4)").textContent.trim();
+                const genderSymbol = item.querySelector("div:nth-child(5)").textContent.trim();
+                const DOB = item.querySelector("div:nth-child(6)").textContent.trim();
+                const DOE = item.getAttribute("data-DOE").trim();
+
+                const maleSymbol = String.fromCharCode(0x2642);
+                const gender = genderSymbol === maleSymbol ? "male" : "female";
+
+                dlForm.querySelector("#documentNumber_dl").value = documentNumber;
+                dlForm.querySelector("#holdingPersonName_dl").value = holdingPersonName;
+                dlForm.querySelector("#DOB_dl").value = convertToDateInputFormat(DOB);
+                dlForm.querySelector(`#gender_${gender}`).checked = true;
+                dlForm.querySelector("#DOE_dl").value = DOE;
+
+                currentEditItem = item;
+            };
+
+            const viewItem = (event) => {
+                const item = event.target.closest(".item");
+                const documentNumber = item.querySelector("div:nth-child(3)").textContent.trim();
+                const holdingPersonName = item.querySelector("div:nth-child(4)").textContent.trim();
+                const genderSymbol = item.querySelector("div:nth-child(5)").textContent.trim();
+                const DOB = item.querySelector("div:nth-child(6)").textContent.trim();
+                const DOE = item.getAttribute("data-DOE").trim();
+
+                generateImage(documentNumber, holdingPersonName, genderSymbol, DOB, DOE);
+            };
+
+            const generateImage = (documentNumber, holdingPersonName, genderSymbol, DOB, DOE) => {
```

Comment:
this must be common method; you just need to pass data and few style.

Reply:
done